src/Common/FileCache.cpp

```diff
@@ -404,9 +406,46 @@ LRUFileCache::FileSegmentCell * LRUFileCache::addCell(
             "Cache already exists for key: `{}`, offset: {}, size: {}.\nCurrent cache structure: {}",
             keyToStr(key), offset, size, dumpStructureUnlocked(key, cache_lock));
 
-    auto file_segment = std::make_shared<FileSegment>(offset, size, key, this, state);
-    FileSegmentCell cell(std::move(file_segment), this, cache_lock);
+    auto skip_or_download = [&]() -> FileSegmentPtr
+    {
+        if (state == FileSegment::State::EMPTY)
+        {
+            LOG_TEST(log, "[addCell] FileSegment key:{}, offset:{}, state:{}, enable_cache_hits:{}, current_element_size:{}/{}.",
```

Comment:
btw I think it will result in too much logging even for test mode. In CI test mode it the default, so CI logs will be overwhelmed. Therefore I believe it is better to remove this logging.

Reply:
It is better to add information about current hits from stash queue to `system.filesystem_cache_log`.